### DISCUSSION TITLE
Transform Hierarchy issue

### DISCUSSION THREAD

**Discussion by Jesh1955** [keiwando/toonsquid#163] [General]
I just got ToonSquid and am excited about using it. I've been going through the tutorials and I'm kind of stuck. I brought two layers into "Transform Hierarchy" and set them up as Parent and Child. For some reason when I move the Parent layer the child isn't connected.  Also, moving the layer doesn't automatically create a key frame. What am I missing?

I just discovered it works fine with vector objects.

**Comment by Cat-Butler** [comment]
Good observations!
It looks like it should be currently possible to make a vector object (as another layer) with no opacity (completely transparent), parent the raster child to that vector object/layer, then animate the vector layer (if you need the visual raster properties, otherwise it would be better to just use the vector object directly).

**Reply by Jesh1955** [reply]
Thanks for the workaround! I'll give it a try.

**Reply by Cat-Butler** [reply]
Hey @Jesh1955, I played some more, thought you might be interested.

If you tap on a raster-painted layer name in the timeline area, you can change a raster layer to a symbol by selecting "Create Symbol" from the menu that pops up (Looks like maybe a standard layer is in "edit" mode, and making it a symbol "freezes" it? Up until then it's a series of separate drawing objects rather than a single object to be rigged.).

This now-a-symbol-layer can have the pivot point changed and will work as expected in a Hierarchy structure.
Hierarchical non-vector rigging!

**Reply by Jesh1955** [reply]
Your first idea works great! Thanks for outlining the "Create Symbols" approach. This app continues to impress me.

**Reply by Cat-Butler** [reply]
Agreed, ToonSquid is very impressive for an iPad app, in current/existing functionality, interface design, and potential future development.

One more rigging/hierarchy note: 
If you edit a raster brush (duplicate it first to preserve the original), and turn on the Vector Mode switch in the brush settings, you lose significant brush detail/subtlety (all variable opacity, for a start), but the brush stroke counts as a vector object for pivot and hierarchy/rigging purposes.

**Comment by keiwando** [comment]
You probably just need to disable the "Edit pixels" option in the trqnsform tool (at the top).

**Reply by Cat-Butler** [reply]
Oh, yes, thank you! That’s the best solution.

Kicking myself, because already I’ve seen you mention that in other contexts.
